src/components/TooltipController/TooltipController.tsx

```diff
@@ -110,10 +109,11 @@ const TooltipController = ({
   }
 
   useEffect(() => {
-    if (content) {
-      setTooltipContent(content)
-    }
+    setIsHtmlContent(false)
+    setTooltipContent(content)
     if (html) {
+      // html will take precedence
+      setIsHtmlContent(true)
       setTooltipContent(html)
```

Comment:
I have the feeling that we will spend resources (a bit) doing the:

```
setIsHtmlContent(false)
```

and after that:

```
setIsHtmlContent(true)
```

In this case, I prefer not to do the `setIsHtmlContent(false)` if we will do `setIsHtmlContent(true)` two lines below.

Is this necessary in this order? or can we add a validation to do one or other?

Reply:
The validation would only serve to make the code clearer (which I think is unnecessary, since it's simple enough).

React 18 batches `useState()` updates, so the state is effectively only updated once. [This source](https://blog.bitsrc.io/automatic-batching-in-react-18-what-you-should-know-d50141dc096e#:~:text=React%20v18%20ensures%20that%20state%20updates%20invoked%20from%20any%20location%20will%20be%20batched%20by%20default.) is a bit outdated, but I'm pretty sure that still holds true.

We can come back to this in the future if we ever decide to improve the `content`/`html` props (maybe there shouldn't be a single state variable - `tooltipContent` - that holds the value for the one being actually used).

Reply:
oh, interesting!

ok